### DISCUSSION TITLE
Role Interface for create custom roles

### DISCUSSION THREAD

**Discussion by lthoa** [oqtane/oqtane.framework#2726] [Ideas] [answered]
Some custom modules need to create their own roles, and it must be added before the CreateSite method is executed because the page templates will use the previously created roles.

**Comment by sbwalker** [comment, marked as answer]
Why not simply create your custom roles at the top of your CreateSite() method? The idea behind the CreateSite() method is that you can execute whatever custom logic you need for provisioning a site.
